src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+    if (this.health <= 0) {
+      this.health = 0;
+      this.remove();
+    }
+  }
+
+  remove() {
+    const index = Animal.alive.indexOf(this);
+    Animal.alive.splice(index, 1);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
```

Comment:
The `hide` method sets `this.hidden` to `true` correctly according to the requirements. However, ensure that it does not toggle the value between `true` and `false` if the method is called multiple times. As per the checklist, the `hide` method should only set the hidden property to `true`.